arcgispro/src/coregister/esri/toolboxes/3D Data Co-Registration.pyt

```diff
@@ -549,10 +551,56 @@ class Register_MultiType(object):
             if parameters[5].value <= parameters[6].value:
                 parameters[5].setErrorMessage("Strong filter size must be larger than weak filter size")
                 parameters[6].setErrorMessage("Weak filter size must be smaller than large filter size")
+        
+        #Check if input DEMs have equal X and Y cell size values
+
+        #first make sure both params are input
+        if parameters[0].value and parameters[1].value:
+            #get path
+            fnd_full_path = os.fsdecode(f"{parameters[0].valueAsText}").replace(os.sep, "/")
+            aoi_full_path = os.fsdecode(f"{parameters[1].valueAsText}").replace(os.sep, "/")
+
+            inputs_list =[fnd_full_path, aoi_full_path]
+            #check for both FND and AOI
+            for input_file in inputs_list:
+                #analysis can only be done with raster/DEM input
+                if os.path.splitext(input_file)[-1] in {'.tif', '.tiff'}:
+                    #see number of bands in raster (Valid DEMs only have 1)
+                    banddesc = arcpy.Describe(input_file)
+                   
+                    #set warning (tool can still be run) if more than one band
+                    if banddesc.bandCount != 1:
+                        if input_file == fnd_full_path:
+                                warningString = f"Warning: Input DEM has more than one band in {os.path.split(fnd_full_path)[-1]}. The tool will not run properly with the input data as is. Consider regenerating input DEM"
+                                parameters[0].setWarningMessage(warningString)
+                        if input_file == aoi_full_path:
+                                warningString = f"Warning: Input DEM has more than one band in {os.path.split(aoi_full_path)[-1]}. The tool will not run properly with the input data as is. Consider regenerating input DEM"
+                                parameters[1].setWarningMessage(warningString)
+                    
+                    #need to access detail of Band1 (or only band for DEMs)
+
+                    #first get band name
+                    arcpy.env.workspace = input_file
+                    bands = arcpy.ListRasters()
+                    #join only band to get band description
+                    DEMband = os.path.join(input_file, bands[0])
+                    desc = arcpy.Describe(DEMband)
```

Comment:
I don't understand what's happening here, or specifically with the line above.  Why is there an `os.path.join` call being done to the input file with the first element of the `bands` list?  That isn't a file that doesn't exist?  I would have thought we would want `desc = arcpy.Describe(input_file)` ?

Reply:
in ArcGIS, it stores the band as /name/of/file.tif/Band1, so the first element of the bands lists gives the only band, then to get the raster band object to apply the .meanCellWidth/Height method attached to the file, the os.path.join is used